File: src/main/java/tn/esprit/spring/model/RegisterDto.java
package tn.esprit.spring.model;


import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import tn.esprit.spring.Entity.Role;

import java.time.LocalDate;


@ToString
@Getter
@Setter
public class RegisterDto {
    private String email;
    private String nom;
    private String prenom;
    private String cin;
    private String password;
    private String telephone;
    private LocalDate dateN ;
    private Role role;
}

File: src/app/components/models/RegisterDto.ts
import { Role } from './Role'; // assuming you have a Role model
export class RegisterDto {
  email: string;
  nom: string;
  prenom: string;
  cin: string;
  password: string;
  telephone: string;
  dateN: Date;
  role: Role;

  constructor(
    email: string,
    nom: string,
    prenom: string,
    cin: string,
    password: string,
    telephone: string,
    dateN: Date,
    role: Role
  ) {
    this.email = email;
    this.nom = nom;
    this.prenom = prenom;
    this.cin = cin;
    this.password = password;
    this.telephone = telephone;
    this.dateN = dateN;
    this.role = role;
  }
}